always-hydrate.md

```diff
@@ -0,0 +1,133 @@
+# Design for always-hydrate feature
+
+## Goal
+The goal of this feature is to enable a developer to communicate that they care about a set of paths,
+and want their IO in those paths to always be fast. E.g. they will always build a certain subfolder,
+and they don't want to pay for file-by-file downloads within that folder.
+
+Today, a user can partially communicate this desire with the `gvfs prefetch --files` or `gvfs
+prefetch --folders` commands to bulk download the files they care about. And they can further specify
+the `--hydrate` flag to either command to bulk hydrate those files into the working directory.
+
+However, this is not a great experience because it's just a one-time prefetch and hydrate. What that
+means is that the next time the projection changes, e.g. after a `git checkout`, anywhere from some
+to all of those files can become dehydrated, and the user will again pay the cost of file-by-file
+downloads while trying to build.
+
+With this feature, we want the developer to make one gesture and communicate that a set of files
```

Comment:
I can see this being helpful, but this would conflict with the way Office plans to consume this feature and would be problematic for build lab scenarios. It would need to have an opt-out flag at least (although I think I would prefer opt-in).

Reply:
See the other thread on this topic, I don't think most users, at least not in Office, will want the behavior you're describing.